### PR TITLE
fix: use anchored regex for NODE_ENV file matching (closes #867)

### LINKED ISSUE #867: Inexact matching in the environment files sanity check identified by codeRabbit
I changed the intentation level of the code which made CR feel entitled to talk smack about the code.

Code Rabbit pointed out that this code will match on env='test' and filename = 'contest.json':

```

    load.options.nodeEnv.forEach(function (env) {
      // Throw an exception if there's no explicit config file for NODE_ENV
      const anyFilesMatchEnv = sourceFilenames.some(function (filename) {
        return filename.match(env);
      });
      // development is special-cased because it's the default value
      if (env && (env !== 'development') && !anyFilesMatchEnv) {
        _warnOrThrow(`${load.getEnv("nodeEnv")} value of '${env}' did not match any deployment config file names.`);
      }
```

It suggested the following edit, which seems pretty close to me.

```
    load.options.nodeEnv.forEach(function (env) {
      // Throw an exception if there's no explicit config file for NODE_ENV
      const anyFilesMatchEnv = sourceFilenames.some(function (filename) {
        return filename.match(new RegExp(`(^|[.-])${env}([.-]|$)`));
      });
      // development is special-cased because it's the default value
      if (env && (env !== 'development') && !anyFilesMatchEnv) {
        _warnOrThrow(`${load.getEnv("nodeEnv")} value of '${env}' did not match any deployment config file names.`);
      }
```
Except that it should probably be:
```
        return filename.match(new RegExp(`^${env}[.-]`));
```
At the very least CodeRabbit failed to match correctly on the file extension. And the groupings are pointless since we're ignoring the return value of `match()`

Because we do, I think, want to declare that 'a test config file was found' when NODE_ENV=test and APP_INSTANCE=1 and we find '/config/test-1.json', but APP_INSTANCE=2 should fail if neither 'test-2.json' nor 'test.json' exist.

But I wanted a sanity check from @markstos or @lorenwest about whether I'm reading this right.

### PR DESCRIPTION
**Pull request by sputnik-mac**
## Problem

The current strictness check uses `filename.match(env)` which performs a substring match. This means `NODE_ENV=test` would falsely match a file named `contest.json`, suppressing the expected warning or exception in strict mode.

## Fix

Replace the substring match with an anchored regular expression:

```js
const envPattern = new RegExp(`^${env}[.-]`);
const anyFilesMatchEnv = sourceFilenames.some(function (filename) {
  return envPattern.test(filename);
});
```

The pattern `^<env>[.-]` ensures:
- The env name appears at the **start** of the filename
- It is immediately followed by a separator (`.` for extension, `-` for instance suffix)
- Partial matches like `contest.json` for `env=test` are correctly rejected

## Tests

- Added `test/6-config/contest.json` as a fixture to represent the false-positive scenario
- Added a regression test: `NODE_ENV=test` in strict mode must throw even when only `contest.json` exists
- Fixed environment state leak in the test helper: `NODE_CONFIG_ENV` is now cleared between suites when not explicitly set, preventing interference between test cases
- All 319 existing tests continue to pass

Fixes #867

## Summary by CodeRabbit

* **Bug Fixes**
  * Enhanced configuration validation matching logic to be more precise and prevent unintended file matches in deployment scenarios.

* **Tests**
  * Added test coverage for configuration validation edge cases to verify correct behavior.

### CODE REVIEW

File: lib/config.mjs [hunk 1]
```diff
@@ -603,9 +603,11 @@ class ConfigUtils {
     });
 
     load.options.nodeEnv.forEach(function (env) {
-      // Throw an exception if there's no explicit config file for NODE_ENV
+      // Throw an exception if there's no explicit config file for NODE_ENV.
+      // Use an anchored regex to avoid false positives (e.g. env='test' matching 'contest.json').
+      const envPattern = new RegExp(`^${env}[.-]`);
       const anyFilesMatchEnv = sourceFilenames.some(function (filename) {
-        return filename.match(env);
+        return envPattern.test(filename);
```
[the remark this answers is not in the document]
Comment (jdmarshall) [reply lib/config.mjs:610]: That would be an “own goal” situation and if you can’t trust your coworkers then you are well and truly doomed.

**Comment by jdmarshall** [comment]
@markstos @lorenwest do we need a policy on AI-generated PRs? 

There’s nothing wrong with this (other than I would have likes a benchmark) and we are using an AI code review tool. So I’m fine with it if you are, but I’m less fine about including agents in the contributors list in the README. It’s difficult enough as a human to get spotted in that wall unless you’re in the top ten. And if AI needs more of that sort of emotional support and gratitude that will be news to me.
